sdk/python/feast/infra/online_stores/contrib/cassandra_online_store/cassandra_online_store.py

```
@@ -623,6 +628,8 @@ def on_failure(exc, concurrent_queue):
             )
             while not concurrent_queue.empty():
                 time.sleep(0.001)
+            if is_error:
+                raise ex
```

You can fail early along with this on line 624, 630